src/pkg/user/service/service.go

```diff
@@ -1,98 +1,400 @@
+		return nil, fmt.Errorf("generateSalt rand.Read error: %w", err)
+	}
+
+	return salt, nil
+}
+
+func (s *Service) generateHash(salt []byte, phrases ...string) ([]byte, error) {
+	password := strings.Join(phrases, "")
+
+	pwdHash, err := scrypt.Key([]byte(password), salt, N, r, p, keyLen)
+	if err != nil {
+		return nil, fmt.Errorf("generateHash scrypt.Key error: %w", err)
+	}
+
+	return pwdHash, nil
+}
+
+func (s *Service) CreateToken(userID string) (*TokenDetails, error) {
+	td := &TokenDetails{}
+	td.AtExpires = time.Now().Add(common.JWTExpires).Unix()
+	td.RtExpires = time.Now().Add(common.JWTRefreshExpires).Unix()
+
+	var err error
+	//Creating Access Token
+	_, accessTokenSecret, err := s.Infra.Keystore.Get(userID)
+	if err != nil {
+		return nil, fmt.Errorf("CreateToken Keystore.Get error: %w userID %s", err, userID)
+	}
+
+	ecdsaAccessKey, err := crypto.ToECDSA(accessTokenSecret[:])
+	if err != nil {
+		return nil, fmt.Errorf("crypto.ToECDSA error: %w userID %s", err, userID)
+	}
+
+	atClaims := jwt.MapClaims{}
+	atClaims["exp"] = td.AtExpires
+	// TODO to fill user metadata like roles we should create new method in contract i.e. UserGet!!!
+	at := jwt.NewWithClaims(jwt.SigningMethodES256, atClaims)
+	td.AccessToken, err = at.SignedString(ecdsaAccessKey)
+
+	if err != nil {
+		return nil, fmt.Errorf("at.SignedString error:%w", err)
+	}
+
+	//Creating Refresh token
+	_, refreshTokenSecret, err := s.Infra.Keystore.Get(userID + "_refresh")
+	if err != nil {
+		return nil, fmt.Errorf("CreateRefreshToken Keystore.Get error: %w userID %s", err, userID)
+	}
+
+	ecdsaRefreshKey, err := crypto.ToECDSA(refreshTokenSecret[:])
+	if err != nil {
+		return nil, fmt.Errorf("crypto.ToECDSA error: %w userID %s", err, userID)
+	}
+
+	rtClaims := jwt.MapClaims{}
+	rtClaims["exp"] = td.RtExpires
+	rt := jwt.NewWithClaims(jwt.SigningMethodES256, rtClaims)
+	td.RefreshToken, err = rt.SignedString(ecdsaRefreshKey)
+
+	if err != nil {
+		return nil, fmt.Errorf("rt.SignedString error:%w", err)
+	}
+
+	return td, nil
+}
+
+func (s *Service) ExtractToken(bearToken string) string {
+	strArr := strings.Split(bearToken, " ")
+	if len(strArr) == 2 {
+		return strArr[1]
+	}
+	return ""
+}
+
+func (s *Service) VerifyToken(userID, tokenString string, isRefreshToken bool) (*jwt.Token, error) {
+	tokenUUID := userID
+	if isRefreshToken {
+		tokenUUID = tokenUUID + "_refresh"
+	}
+
+	_, tokenSecret, err := s.Infra.Keystore.Get(tokenUUID)
+	if err != nil {
+		return nil, fmt.Errorf("VerifyToken Keystore.Get error: %w userID %s", err, userID)
+	}
+
+	ecdsaKey, err := crypto.ToECDSA(tokenSecret[:])
+	if err != nil {
+		return nil, fmt.Errorf("crypto.ToECDSA error: %w userID %s", err, userID)
+	}
+
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodECDSA); !ok {
+			return nil, fmt.Errorf("%w signing method: %v", errors.ErrIsUnsupported, token.Header["alg"])
+		}
+		return ecdsaKey.Public(), nil
```

Comment:
its not required because when we run jwt.Parse() it invoke golang-jwt/jwt@v3.2.2+incompatible/parser.go  -> ParseWithClaims() which contain all checks

![image](https://user-images.githubusercontent.com/4029273/197154173-eb1b9e6d-a320-47bb-aa38-84c13ce6e356.png)